test/inductor/test_torchinductor.py

```diff
@@ -10358,6 +10358,19 @@ def fn(x):
         x = torch.rand([4, 4, 3], dtype=torch.float64)
         self.common(fn, (x,))
 
+    @skipCUDAIf(not SM80OrLater, "uses bfloat16 which requires SM >= 80")
+    def test_dtypeview(self):
+        # https://github.com/pytorch/pytorch/issues/126338
```

Comment:
parameterize test over different dtype inputs and views, needs to not just run the function but test reuslts. see the rest of the tests in this file - e.g., self.common

Reply:
bump

Reply:
After I changed it to `self.common`, there is an error. 
```
torch._dynamo.exc.TorchRuntimeError: Failed running call_function <built-in function matmul>(*(FakeTensor(..., device='cuda:0', size=(2, 4), dtype=torch.float16), FakeTensor(..., device='cuda:0', size=(2, 4), dtype=torch.float16)), **{}):
a and b must have same reduction dim, but got [2, 4] X [2, 4].

from user code:
   File "/home/yhao/p9/pytorch/test/inductor/test_torchinductor.py", line 10368, in fn
    return x @ y
```
I found it is because the `self.common` changed the input type when it is float16. https://github.com/pytorch/pytorch/blob/2213ff44599087a012795536f7742d18fa5b20ec/test/inductor/test_torchinductor.py#L415
That's why I didn't use `self.common`. Is there any way to disable this conversion?

Reply:
`reference_in_float=False`

Reply:
fixed in https://github.com/pytorch/pytorch/pull/128883/commits/9db9195de5b9530e2c49e104ffe7e1bad5c89315

Reply:
Still need to parameterize over different input/output dtypes